packages/node/src/services.ts

```diff
@@ -232,3 +236,102 @@ export function confirmLocalFirebaseConfigurationEnvVars() {
     );
   }
 }
+
+export const POSTGRES_CONFIGURATION_ENV_KEYS = {
+  username: "POSTGRES_USER",
+  host: "POSTGRES_HOST",
+  database: "POSTGRES_DATABASE",
+  password: "POSTGRES_PASSWORD",
+  port: "POSTGRES_PORT"
+};
+
+export const EMPTY_POSTGRES_CONFIG: ConnectionOptions = {
+  type: "postgres",
+  username: "",
+  host: "",
+  database: "",
+  password: "",
+  port: 0
+};
+
+export class PostgresServiceFactory {
+  private connectionManager: ConnectionManager;
+  private connection: Connection;
+
+  constructor(configuration: ConnectionOptions) {
+    this.connectionManager = new ConnectionManager();
+    this.connection = this.connectionManager.create({
+      ...EMPTY_POSTGRES_CONFIG,
+      ...configuration,
+      entities: [NodeEntity]
+    } as ConnectionOptions);
+  }
+
+  async connectDb() {
+    await this.connection.connect();
+  }
+
+  createStoreService(storeServiceKey: string): IStoreService {
+    console.log("Connected to Postgres");
+    return new PostgresStoreService(this.connectionManager, storeServiceKey);
+  }
+}
+
+class PostgresStoreService implements IStoreService {
+  constructor(
+    private readonly connectionMgr: ConnectionManager,
+    private readonly storeServiceKey: string
+  ) {}
+
+  async set(
+    pairs: { key: string; value: any }[],
+    allowDelete?: Boolean
+  ): Promise<boolean> {
+    if (!allowDelete && containsNull(pairs)) {
+      throw new Error(WRITE_NULL_TO_FIREBASE);
+    }
+
+    const connection = this.connectionMgr.get();
+
+    await connection.transaction(async transactionalEntityManager => {
+      for (const pair of pairs) {
+        const storeKey = `${this.storeServiceKey}_${pair.key}`;
+        // wrap value so its always JSON
```

Comment:
This is necessary for Postgres because the JSON column breaks if you use anything other than JSON (i.e. a raw string). In some cases, the node code is inserting strings as values instead of objects.